WordPress/Classes/ViewRelated/Activity/ActivityDetailViewController.swift

```diff
@@ -169,5 +176,46 @@ class ActivityDetailViewController: UIViewController {
     private enum Constants {
         static let gridiconSize: CGSize = CGSize(width: 24, height: 24)
     }
+}
+
+// MARK: - Navigation
+
+extension ActivityDetailViewController {
+    private func routeTo(_ url: URL) {
```

Comment:
Would it make sense to extract this code to a separate entity? Being in an extension does not actually extract the code from the view controller, and it seems like there are no direct dependencies between this code and the rest of the view controller.

I think there might be another benefit as well: if this is extracted, then the view controller would not need to be aware of the `Router`, which might help decouple it from navigation. 

That way, as well, we could migrate to the routing system @frosty introduced if/when necessary. What do you think?

Reply:
100%!
I basically kept it the same way it was on NotificationDetails. This means that we can do similar a improvement there too :)

Any name ideas for that new entity?

Reply:
I'm terrible at naming 😛 All I can think of is `NotificationNavigationCoordinator` (facepalm)